.github/workflows/integration-tests.yml.in

```diff
@@ -384,3 +390,50 @@ jobs:
       - *run-cpp-unittests-step
       - *save-build-artifacts-step
       - *inspect-cache-directories-step
+
+  Build-Tests:
+    needs: Runner-Preparation
+    if: needs.Runner-Preparation.outputs.matrix-MACOS != ''
+    runs-on: ${{ matrix.runner }}
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        runner: ${{fromJson(needs.Runner-Preparation.outputs.matrix-MACOS)}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: "true"
+      - name: Install brew dependencies
+        run: |
+          brew update
+          brew install ccache
+          brew install llvm
+
+      - *compute-cache-keys-step
+      - *cache-build-dependencies-step
+      - *restore-build-artifacts-step
+      - *inspect-cache-directory-step
+
+      - name: Update PATH
+        run: |
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+          echo "/opt/homebrew/opt/llvm/bin" >> $GITHUB_PATH
+      - name: Install pip dependencies
+        run: |
+          python3 -m venv ~/.venv
+          source ~/.venv/bin/activate
+          python3 -m pip install --upgrade pip
+          python3 -m pip install cython setuptools wheel cmake==3.24 ninja pytest-xdist lit
+      - name: Install Triton
+        env:
+          TRITON_BUILD_WITH_CCACHE: "true"
+          TRITON_BUILD_WITH_O1: "true"
+          # macos-latest has a 3-core M1 vcpu
+          # https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories
+          MAX_JOBS: 3
```

Comment:
I don't think this is quite the right comment.  `ninja -j` will read the number of CPUs and launch f(ncpu) jobs, I think 2*ncpu.

You're restricting to 3 jobs and because I presume you found it's faster?  Probably due to limited RAM?

Reply:
Yes. Due to limited RAM.

I'd like to launch `ncpu` instead of `2*cpu` because M1 doesn't have hyperthreading.

Reply:
Right, just suggest updating the comment to explain the reason, which isn't really that M1 only has 3 cores.

(Also usually build systems will do 2*ncpu even if there's no hyperthreading, because sometimes jobs are I/O bound.  Indeed I think they probably do 2*nvcpu, i.e. if M1 had hyperthreading it would by default launch 12 jobs.)